config/config.go

```diff
@@ -1,16 +1,70 @@
 package config
 
-import "time"
+import (
+	ldbopt "github.com/syndtr/goleveldb/leveldb/opt"
+)
 
-// ReplicaConfig specifies the information needed for a Replica and Version manager to
-// maintain global object versioning and provenance.
-// TODO: this configuration is pulled from a service context not a library context.
+// DefaultConfig is used if the user does not specify a configuration
+var DefaultConfig = Config{
+	Versions: ReplicaConfig{
+		PID:    1,
+		Region: "local",
+		Name:   "localhost",
+	},
+}
+
+// New creates a configuration with the required options and can also be used to specify
+// optional configuration e.g. for engine-specific operations.
+func New(options ...Option) Config {
+	// Create the default configuration in editable mode
+	conf := &Config{
+		Versions: ReplicaConfig{
+			PID:    DefaultConfig.Versions.PID,
+			Region: DefaultConfig.Versions.Region,
+			Name:   DefaultConfig.Versions.Name,
+		},
+	}
+
+	// Apply all options to the configuration
+	for _, opt := range options {
+		opt(conf)
+	}
+
+	// Return the value of the configuration
+	return *conf
+}
+
+// Config specifies the options necessary to open a Honu database.
+type Config struct {
+	Versions   ReplicaConfig
+	LDBOptions *ldbopt.Options
+}
+
+// ReplicaConfig specifies the information needed for the Version manager to maintain
+// global object versioning and provenance. This is closely tied to a Replica's
+// configuration, e.g. the PID is the process ID of a running replica, the region is
+// where the replica is running, and the name is usually the hostname of the replica.
 type ReplicaConfig struct {
-	Enabled        bool          `split_words:"true" default:"true"`
-	BindAddr       string        `split_words:"true" default:":4435"`
-	PID            uint64        `split_words:"true" required:"false"`
-	Region         string        `split_words:"true" required:"false"`
-	Name           string        `split_words:"true" required:"false"`
-	GossipInterval time.Duration `split_words:"true" default:"1m"`
-	GossipSigma    time.Duration `split_words:"true" default:"5s"`
+	PID    uint64 `split_words:"true" required:"false"`
+	Region string `split_words:"true" required:"false"`
+	Name   string `split_words:"true" required:"false"`
+}
```

Comment:
This is the minimum set of information that the `VersionManager` needs. If Honu becomes ninja trtl, this will need to be refactored once again; probably back to the original version. However, the entire configuration system will also have to be refactored anyway.

Reply:
I think this makes sense for now. We might need to do a bit more work to explain what a replica is inside the context of Honu, since it might not make sense to someone coming to the project without knowledge of GDS and Trtl